vendor/github.com/containers/image/docker/docker_client.go

```diff
@@ -146,21 +156,22 @@ func (c *dockerClient) setupRequestAuth(req *http.Request) error {
 			// no need for bearer? wtf?
 			return nil
 		}
-		tokens = strings.Split(hdr, " ")
-		tokens = strings.Split(tokens[1], ",")
-		var realm, service, scope string
-		for _, token := range tokens {
-			if strings.HasPrefix(token, "realm") {
-				realm = strings.Trim(token[len("realm="):], "\"")
-			}
-			if strings.HasPrefix(token, "service") {
-				service = strings.Trim(token[len("service="):], "\"")
-			}
-			if strings.HasPrefix(token, "scope") {
-				scope = strings.Trim(token[len("scope="):], "\"")
-			}
+		re := regexp.MustCompile(`^Bearer realm="(.*)",service="(.*)",scope="(.*)"$`)
```

Comment:
This is replacing one fragile non-conforming parser with another fragile non-conforming parser. Isn’t there a library which does this truly correctly? (I couldn’t find one in a quick check.)

I do realize that a working fragile parser is better than a non-working fragile parser, sure :)  At the very least, please use `"([^"]*)"` instead of `"(.*)"` to avoid ambiguity.

It would be nice to have a test to know when this fragile parser breaks, though that is non-blocking, better an improvement without a test than broken code.

(Also, doing the regexp compilation once at module load time instead of every time would be nice.)